src/hotspot/share/classfile/classFileParser.hpp

```diff
@@ -461,6 +461,7 @@ class ClassFileParser {
                                     TRAPS) const;
   int  verify_legal_method_signature(const Symbol* methodname,
                                      const Symbol* signature,
+                                     bool check_compatibility,
```

Comment:
Can't help but think this method is trying to perform two jobs when it should only be doing one. What we previously considered "legal" is not actually a check of legality per-se as it is context specific. Perhaps this should really be split into two sets of checks?

Reply:
Hi David, please review the latest commit that breaks verify_legal_method_signatures() into two functions, one that checks that the signature is legal, and the other, verify_legal_name_with_signature(), that checks that the method name and signature are compatible with each other.